x/eibc/keeper/msg_server.go

```diff
@@ -72,6 +72,125 @@ func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrde
 	return &types.MsgFulfillOrderResponse{}, nil
 }
 
+func (m msgServer) FulfillOrderAuthorized(goCtx context.Context, msg *types.MsgFulfillOrderAuthorized) (*types.MsgFulfillOrderAuthorizedResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := ctx.Logger()
+
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	demandOrder, err := m.GetOutstandingOrder(ctx, msg.OrderId)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := m.validateOrder(demandOrder, msg, ctx); err != nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrUnauthorized, err.Error())
+	}
+
+	granterAccount := m.ak.GetAccount(ctx, msg.GetGranterBech32Address())
+	if granterAccount == nil {
+		return nil, types.ErrGranterAddressDoesNotExist
+	}
+
+	// Send the funds from the granterAccount to the eibc packet original recipient
+	err = m.bk.SendCoins(ctx, granterAccount.GetAddress(), demandOrder.GetRecipientBech32Address(), demandOrder.Price)
+	if err != nil {
+		logger.Error("Failed to send price to recipient", "error", err)
+		return nil, err
+	}
+
+	fulfillerAccount := m.ak.GetAccount(ctx, msg.GetFulfillerBech32Address())
+	if fulfillerAccount == nil {
+		return nil, types.ErrFulfillerAddressDoesNotExist
+	}
+
+	feePartReceiver := fulfillerAccount
+	if msg.OperatorAddress != "" {
+		operatorAccount := m.ak.GetAccount(ctx, msg.GetOperatorBech32Address())
+		if operatorAccount == nil {
+			return nil, types.ErrOperatorAddressDoesNotExist
+		}
+		feePartReceiver = operatorAccount
+	}
+
+	fee, _ := sdk.NewDecFromStr(msg.ExpectedFee)
```

Comment:
the expected fee should be provided in percentage, should be > min_fee of the LP and once we validated it, than the operator can take his fee (which is a percentage of the earnings of the LP)

Reply:
the expected fee is the same as in the original `MsgFulfillOrder`, we need to pass it in order to verify that the fulfilled is aware of the fee of the demand order, so it cannot be percentage.
I'm just using it here for convenience in converting to decimal, so I can calculate the operator fee.
I could have also just done `fee, _ := sdk.NewDecFromStr(demandOrder.Fee.String())` instead.

It is already compared to the order.Fee, as the order.Fee is not percentage